iidm/iidm-converter-api/src/main/java/com/powsybl/iidm/AbstractOptions.java

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright (c) 2019, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.iidm;
+
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * @author Chamseddine BENHAMED <chamseddine.benhamed at rte-france.com>
+ */
+public abstract class AbstractOptions<T> {
+    protected Set<String> extensions = null;
+
+    protected IidmImportExportMode mode = IidmImportExportMode.UNIQUE_FILE;
+
+    public AbstractOptions() {
```

Comment:
this constructor is not useful

Reply:
> When I do:
> 
> ```java
> try (FileOutputStream os = new FileOutputStream("/tmp/os.xiidm")) {
>             NetworkXml.write(MultipleExtensionsTestNetworkFactory.create(), new ExportOptions().setMode(IidmImportExportMode.ONE_SEPARATED_FILE_PER_EXTENSION_TYPE), os);
>         }
> ```
> 
> the behaviour of the method is not the expected one: everything is written in one unique file.

Alright, this method should be used only for the unique file mode. i'll add and exception if one of the other modes is used.